File: code/java/productExceptSelf.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] res = new int[nums.length];
        res[0] = 1;
        for(int i=1;i<nums.length;i++)
            res[i] = res[i-1] * nums[i-1];
        int temp = 1;
        for(int i=0;i<nums.length;i++){
            res[nums.length-1-i] *= temp;
            temp *= nums[nums.length-1-i];
        }
        return res;
    }
}

File: code/python/productExceptSelf.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = []
        for i in range(len(nums)):
            res.append(1)
        for i in range(1,len(nums)):
            res[i] *= (res[i-1]*nums[i-1])
 
        temp = 1
        for i in range(0,len(nums)):
            res[len(nums)-i-1] *= temp
            temp *= nums[len(nums)-1-i]
            
        return res
